packages/editor/src/components/save-publish-panels/index.js

```diff
@@ -54,7 +62,7 @@ export default function SavePublishPanels( {
 				PostPublishExtension={ PluginPostPublishPanel.Slot }
 			/>
 		);
-	} else if ( hasNonPostEntityChanges ) {
+	} else if ( hasNonPostEntityChanges || hasPostMetaChanges ) {
```

Comment:
To be honest, I don't fully understand what this part of the code does and if it is necessary or not.

Reply:
As far as I remember, this alternative flow triggers when you edit the template of the post. Let me record a screencast to better illustrate it.

Reply:
https://github.com/WordPress/gutenberg/assets/699132/da45473c-9800-449d-aeed-06d4ad91f4d6

Reply:
In effect, what I think we should achieve with this PR would be:
- regular post - Publish
- regular post with post meta from the same post - Publish (covers Footnotes)
- regular post with any changes to external post - Save + Publish flow

Reply:
To support my line of thinking, updating the post meta of the currently edited post, isn't a site change. What I see currently after adding footnotes to the published post:

<img width="1197" alt="Screenshot 2024-05-31 at 09 55 38" src="https://github.com/WordPress/gutenberg/assets/699132/bd20d34e-c5bc-4c79-9c0a-8979b22fd38c">

Reply:
Okay! Thanks a lot for the video 🙂 I believe I have now a better idea of what that part of the code is doing.

If I understand you correctly, your suggestions are aligned with the use cases I stated in [this comment](https://github.com/WordPress/gutenberg/pull/61811#issuecomment-2139757254), right?

Reply:
For the third case, it seems to be pretty close in terms of the flow, but there is no mention that post meta has changed:

<img width="1375" alt="Screenshot 2024-05-31 at 10 11 14" src="https://github.com/WordPress/gutenberg/assets/699132/08c2b45b-14b8-45b3-bdb6-351dd348d8c5">

Reply:
When I also apply modification to the currently edited post, I see that the annotation about post meta gets applied to all modified posts:

<img width="1365" alt="Screenshot 2024-05-31 at 10 14 00" src="https://github.com/WordPress/gutenberg/assets/699132/c25ef49a-87fe-4e59-8b6e-a005d317eabf">

It means that the check for post meta edits gets set to `true` only when the currently edited post is modified, and in other instances incorrect assertion is used.

Reply:
For completeness, this is what I'm seeing in the site editor:

<img width="1371" alt="Screenshot 2024-05-31 at 10 20 55" src="https://github.com/WordPress/gutenberg/assets/699132/75783601-7873-4990-8da0-549b8977a91e">

Reply:
If I am not mistaken, that is caused because the selector gets the type and id using `getCurrentPost`: [link](https://github.com/WordPress/gutenberg/pull/61811/files#diff-a1cdb9f3f4ce98ee4ac08627daff3c4cb003766468d4bb3780fe0be22a0ad87eR150). Maybe we need to pass the type and id as an argument. I can give that a try.

Reply:
In my opinion, it should use the selector from the `core` store directly so the post id and post type isn't connected with the currently edited entity - post, page, template, etc.

Reply:
Do you mean using `getEntityRecordNonTransientEdits` directly and removing the `hasPostMetaChanges` one we are creating?

Reply:
I pushed [this commit](https://github.com/WordPress/gutenberg/pull/61811/commits/baf5ff7973024210315b6991c8cfb1ca474a646e) to illustrate what I mean, but I am happy to iterate once I understand better your suggestion.

If I am not mistaken, that change fixes the problems you mention.

Reply:
I tested @SantosGuillamot's fix for to address the scenario wherein the indicator for post meta changes didn't appear when modifying posts in the query loop. It appears to work:

<img width="842" alt="Screenshot 2024-05-31 at 12 45 10 PM" src="https://github.com/WordPress/gutenberg/assets/5360536/a21d8a20-e4a7-44f5-bf30-9b96116d3526">